meson.build

```diff
@@ -30,6 +30,27 @@ project('sentry-kernel', 'c',
 
 meson.add_dist_script('support/meson/version.sh', 'set-dist', meson.project_version())
 
+# Testing high security flags of cross compiler. These are gcc 13-14 hardening flags.
+# See https://gcc.gnu.org/onlinedocs/gcc-14.1.0/gcc/Instrumentation-Options.html#index-fharden-compares
+# for more information about each flag
+hardening_cflags = [
+  '-fharden-compares',
+  '-fharden-conditional-branches',
+  '-fharden-control-flow-redundancy',
+  '-fhardcfr-check-returning-calls',
+  '-fstack-clash-protection',
+]
+activated_hardening_cflags = []
+
+compiler = meson.get_compiler('c', native: false)
+
+foreach cflag: hardening_cflags
+  if compiler.has_argument(cflag)
+    activated_hardening_cflags += cflag
+  endif
+endforeach
```

Comment:
Is this kconfig enabled feature ?
For instance `config SECU_HARDEN_COMPILE_FLAGS`, forced in release, but 'selectable' in other build flavor.
the same as 
```kconfig
config SECU_SSP_USER
	bool
```
(may be in a next PR)

Reply:
note: wa already talk about that for a subset of kernel (available since there is interna static libs for modules)